### PR TITLE
Add extensions to context.go for introspection features

### PR DESCRIPTION
**Pull request by jin**
See https://android-review.googlesource.com/c/platform/build/soong/+/1359245 for context

### CODE REVIEW

**Comment by paulduffin** [comment]
The TargetNameWithVariant() and QualifiedTargetLabel() methods do not appear to need to be in Blueprint as they only use capabilities that are available through the existing interfaces. They are also quite specific to the use being made of them inside Soong (the Android build system) and so should be part of that not Blueprint. You will also find that there is already some similar code used in the visibility functionality in Soong which was based, at least in part, on the Bazel support for visibility.

**Comment by jin** [comment]
@paulduffin Thank you! Good idea.

File: context.go [hunk 1]
```diff
@@ -3071,6 +3071,22 @@ func (c *Context) BlueprintFile(logicModule Module) string {
 	return module.relBlueprintsFile
 }
 
+func (c *Context) LogicModules() []Module {
```
Comment (colincross) [inline context.go:3074]: This is already available through ctx.VisitAllModules.

File: context.go [hunk 2]
```diff
@@ -3071,6 +3071,22 @@ func (c *Context) BlueprintFile(logicModule Module) string {
...
+	modules := []Module{}
+	for k, _ := range c.moduleInfo {
+		modules = append(modules, k)
+	}
+	return modules
+}
+
+func (c *Context) DirectDeps(logicModule Module) []Module {
```
Comment (colincross) [inline context.go:3082]: This is already available through ctx.VisitDrirectDeps.

File: context.go (hunk 1 above)
Comment (jin) [reply context.go:3074]: Ahh, of course! Thank you, this PR is not needed then :)

**Pull request closed without merging**